orchestrator/main_loops.go

```diff
@@ -280,15 +281,34 @@ func (p *peggyOrchestrator) CheckFeeThreshold(
 		return true
 	}
 
+	decimals, err := p.peggyContract.GetERC20Decimals(ctx, erc20Contract, p.peggyContract.FromAddress())
+	if err != nil {
+		p.logger.Err(err).Str("token_contract", erc20Contract.String()).Msg("failed to get token decimals")
+		return false
+	}
+
+	p.logger.Debug().
+		Uint8("decimals", decimals).
+		Str("token_contract", erc20Contract.String()).
+		Msg("got token decimals")
+
 	tokenPriceInUSD, err := p.priceFeeder.QueryUSDPrice(erc20Contract)
 	if err != nil {
 		return false
 	}
 
 	tokenPriceInUSDDec := decimal.NewFromFloat(tokenPriceInUSD)
-	totalFeeInUSDDec := decimal.NewFromBigInt(totalFee.BigInt(), -18).Mul(tokenPriceInUSDDec)
+	totalFeeInUSDDec := decimal.NewFromBigInt(totalFee.BigInt(), -int32(decimals)).Mul(tokenPriceInUSDDec)
```

Comment:
Should we be safe and return an error on overflow? i.e. return an error if `decimals > math.MaxInt32`?

Reply:
Gonna add that 👌

Reply:
Ah, we don't need it, decimals is a uint8 (even in the ERC20 contract is being stored as a uint8).

Reply:
Let's add a comment then 👍